subsys/net/buf.c

```diff
@@ -357,6 +361,37 @@ struct net_buf *net_buf_alloc_fixed(struct net_buf_pool *pool, s32_t timeout)
 }
 #endif
 
+#if defined(CONFIG_NET_BUF_LOG)
+struct net_buf *net_buf_alloc_with_data_debug(struct net_buf_pool *pool,
+					      void *data, size_t size,
+					      s32_t timeout, const char *func,
+					      int line)
+#else
+struct net_buf *net_buf_alloc_with_data(struct net_buf_pool *pool,
+					void *data, size_t size,
+					s32_t timeout)
+#endif
+{
+	struct net_buf *buf;
+
+#if defined(CONFIG_NET_BUF_LOG)
+	buf = net_buf_alloc_len_debug(pool, 0, timeout, func, line);
+#else
+	buf = net_buf_alloc_len(pool, 0, timeout);
+#endif
+	if (!buf) {
+		return NULL;
+	}
+
+	buf->__buf = data;
```

Comment:
It looks like you're missing `buf->data = buf->__buf;`

Reply:
Or `buf->data = data;` to be consistent with how you're initializing `buf->size` and `buf->len` from the same `size` variable.

Reply:
I was sure net_buf_reset() in alloc len did it... yes with a NULL pointer, obviously. ^^'